main.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"encoding/json"
+
+	"github.com/mount-joy/thelist-lambda/handlers"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+type handler struct {
+	router handlers.Router
+}
+
+func (h *handler) doRequest(request events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
+	result, statusCode := h.router.Route(request)
+
+	var res []byte
+	if result != nil {
+		var err error
+		res, err = json.Marshal(result)
+		if err != nil {
+			return events.APIGatewayProxyResponse{
+				Body:       err.Error(),
+				StatusCode: 500,
+			}, nil
+		}
+	}
```

Comment:
This part seems a bit confusing to me, if `result == nil` we end up with an empty body (which isn't valid json), but we're still sending a response content type of `application/json`:

```bash
$ curl -vs http://127.0.0.1:3000/hi 2>&1 | grep Content-
< Content-Type: application/json
< Content-Length: 0
```

However, if we remove the `result != nil` check:

```diff
diff --git a/main.go b/main.go
index 5dccf86..fb7cf93 100644
--- a/main.go
+++ b/main.go
@@ -15,17 +15,13 @@ type handler struct {
 
 func (h *handler) doRequest(request events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
 	result, statusCode := h.router.Route(request)
+	res, err := json.Marshal(result)
 
-	var res []byte
-	if result != nil {
-		var err error
-		res, err = json.Marshal(result)
-		if err != nil {
-			return events.APIGatewayProxyResponse{
-				Body:       err.Error(),
-				StatusCode: 500,
-			}, nil
-		}
+	if err != nil {
+		return events.APIGatewayProxyResponse{
+			Body:       err.Error(),
+			StatusCode: 500,
+		}, nil
 	}
 
 	return events.APIGatewayProxyResponse{

```

Then we get `null` as the body which is at least valid json. Alternatively we could change the response content-type.

Reply:
Yeah good point - will change